hal/serial_api.h

```diff
@@ -108,9 +110,90 @@ extern "C" {
 
 /**
  * \defgroup hal_GeneralSerial Serial Configuration Functions
+ *
+ * # Defined behavior
+ * * ::serial_init initializes the ::serial_t
+ * * ::serial_init sets the default parameters for serial peripheral (9600 bps, 8N1 format)
+ * * ::serial_init configures the specified pins
+ * * ::serial_free releases the serial peripheral
+ * * ::serial_baud configures the baud rate
+ * * at least 9600 bps the baud rate  must be supported
+ * * ::serial_format configures the transmission format (number of bits, parity and the number of stop bits)
+ * * at least 8N1 format must be supported
+ * * ::serial_irq_handler registers the interrupt handler which will be invoked when the interrupt fires.
+ * * ::serial_irq_set enables or disables the serial RX or TX IRQ.
+ * * If `RxIrq` is enabled by ::serial_irq_set, ::serial_irq_handler will be invoked whenever
+ * Receive Data Register Full IRQ is generated.
+ * * If `TxIrq` is enabled by ::serial_irq_set, ::serial_irq_handler will be invoked whenever
+ * Transmit Data Register Empty IRQ is generated.
+ * * If the interrupt condition holds true, when the interrupt is enabled with ::serial_irq_set,
+ * the ::serial_irq_handler is called instantly.
+ * * ::serial_getc returns the character from serial buffer.
+ * * ::serial_getc is a blocking call (waits for the character).
+ * * ::serial_putc sends a character.
+ * * ::serial_putc is a blocking call (waits for a peripheral to be available).
+ * * ::serial_readable returns non-zero value if a character can be read, 0 otherwise.
+ * * ::serial_writable returns non-zero value if a character can be written, 0 otherwise.
+ * * ::serial_clear clears the ::serial_t RX/TX buffers
+ * * ::serial_break_set sets the break signal.
+ * * ::serial_break_clear clears the break signal.
+ * * ::serial_pinout_tx configures the TX pin as an output (to be used in half-duplex mode).
+ * * ::serial_set_flow_control configures serial flow control.
+ * * ::serial_set_flow_control sets flow control in the hardware if a serial peripheral supports it,
+ * otherwise software emulation is used.
+ * * ::serial_tx_asynch starts the serial asynchronous transfer.
+ * * ::serial_tx_asynch writes `tx_length` bytes from the `tx` to the bus.
+ * * ::serial_tx_asynch must support 8 bits words
```

Comment:
```suggestion
 * * ::serial_tx_asynch ignores the `tx_width` argument
```
I don't understand the part with 8 bit words. Maybe we should go back to the original info about arg deprecation?

Reply:
I decided to remove the requirement for ignoring the argument.
The `must support 8 bits words` requirement is misleading. Should be   `must support 8 data bits`.

Is this proposition ok?

Reply:
Yeah, that's ok.

Reply:
This is the last thing that needs an update.